### PR TITLE
ffmpeginput

### PR DESCRIPTION
**Pull request by fafik23**
FIX ffmpeginput to read all frame. Tested with ffmpeg-2.8.1

### CODE REVIEW

File: src/ffmpeg.imageio/ffmpeginput.cpp [hunk 1]
```diff
@@ -206,29 +213,31 @@ FFmpegInput::open (const std::string &name, ImageSpec &spec)
         !strcmp (m_codec_context->codec->name, "dvvideo")) {
         m_offset_time = false;
     }
+
+   m_codec_cap_delay=m_codec_context->codec->capabilities&AV_CODEC_CAP_DELAY ;
```
Comment (lgritz) [inline src/ffmpeg.imageio/ffmpeginput.cpp]: In the version of ffmpeg on one of my systems, AV_CODEC_CAP_DELAY is not found.

But there is a CODEC_CAP_DELAY. Are they the same, and I'm using a slightly older version, after which the AV_ was prepended?

do you think it's sufficient if I just add something like:

    #ifndef AV_CODEC_CAP_DELAY
    #define AV_CODEC_CAP_DELAY CODEC_CAP_DELAY
    #endif

Comment (fafik23) [inline src/ffmpeg.imageio/ffmpeginput.cpp]: We can swich AV_CODEC_CAP_DELAY to CODEC_CAP_DELAY.
https://ffmpeg.org/doxygen/2.8/libavcodec_2avcodec_8h_source.html#l01064




2015-12-10 19:08 GMT+01:00 Larry Gritz <notifications@github.com>:

Comment (lgritz) [inline src/ffmpeg.imageio/ffmpeginput.cpp]: OK, I'll make that change as part of the merge.

**Comment by lgritz** [comment]
Can you explain briefly what problem this addresses? Is this related to issue #1003? Or is it something else?

**Comment by fafik23** [comment]
first it resolved problem with codec AV_CODEC_CAP_DELAY
http://www.ffmpeg.org/doxygen/2.8/group__lavc__decoding.html#ga99ee61b6dcffb7817a275d39da58cc74

second i remove condition to run avcodec_decode_video2 this make that we
put to codec only frame with pts >= pts we looking thats why we see only
key frames.
if (current_pos >= m_subimage) {
avcodec_decode_video2 (m_codec_context, m_frame, &finished, &pkt)
}

Last I add if( current_frame == frame && finished) to be sure we display
exactly this frame we need (current_frame is take from m_frame->pkt_pts
http://www.ffmpeg.org/doxygen/2.8/structAVFrame.html#a47f555732f9c18928de727a163cd7cb6
). this can make infinity loop when something go wrong.

I have not  tested this on codec with bi-directional frame.


2015-12-10 19:40 GMT+01:00 Larry Gritz <notifications@github.com>:

**Comment by lgritz** [comment]
Thanks, I'll add that to the commit message before I merge.

**Comment by lgritz** [comment]
Merged into master, with the fix described above and also a few very small formatting changes.

I'll eventually merge this into RB-1.6, but we're trying to lock it down for a release right now, so I don't want to merge it there until after the beta is over and a release is tagged. That will also give people a little while to test this further in master.

**Pull request closed without merging**